kedro/framework/cli/micropkg.py

```diff
@@ -0,0 +1,293 @@
+"""A collection of CLI commands for working with Kedro micro-packages."""
+
+import re
+import sys
+import tempfile
+from pathlib import Path
+
+import click
+
+from kedro.framework.cli.pipeline import (
+    _append_package_reqs,
+    _assert_pkg_name_ok,
+    _check_module_path,
+    _check_pipeline_name,
+    _find_config_files,
+    _generate_wheel_file,
+    _get_default_version,
+    _get_pipeline_artifacts,
+    _install_files,
+    _unpack_wheel,
+    _validate_dir,
+)
+from kedro.framework.cli.utils import (
+    KedroCliError,
+    _clean_pycache,
+    _get_requirements_in,
+    command_with_verbosity,
+    env_option,
+)
+from kedro.framework.startup import ProjectMetadata
+
+_SETUP_PY_TEMPLATE = """# -*- coding: utf-8 -*-
+from setuptools import setup, find_packages
+
+setup(
+    name="{name}",
+    version="{version}",
+    description="Modular pipeline `{name}`",
+    packages=find_packages(),
+    include_package_data=True,
+    package_data={package_data},
+    install_requires={install_requires},
+)
+"""
+
+
+# pylint: disable=missing-function-docstring
+@click.group(name="Kedro")
+def micropkg_cli():  # pragma: no cover
+    pass
+
+
+@micropkg_cli.group()
+def micropkg():
+    """Commands for working with micro-packages."""
+
+
+@command_with_verbosity(micropkg, "pull")
+@click.argument("package_path", nargs=1, required=False)
+@click.option(
+    "--all",
+    "-a",
+    "all_flag",
+    is_flag=True,
+    help="Pull and unpack all micro-packages in the `pyproject.toml` package manifest section.",
+)
+@env_option(
+    help="Environment to install the micro-package configuration to. Defaults to `base`."
+)
+@click.option(
+    "--alias",
+    type=str,
+    default="",
+    callback=_check_pipeline_name,
+    help="Alternative name to unpackage under.",
+)
+@click.option(
+    "--fs-args",
+    type=click.Path(
+        exists=True, file_okay=True, dir_okay=False, readable=True, resolve_path=True
+    ),
+    default=None,
+    help="Location of a configuration file for the fsspec filesystem used to pull the package.",
+)
+@click.pass_obj  # this will pass the metadata as first argument
+def pull_package(  # pylint:disable=unused-argument, too-many-arguments
+    metadata: ProjectMetadata, package_path, env, alias, fs_args, all_flag, **kwargs
+) -> None:
+    """Pull and unpack a modular pipeline in your project."""
```

Comment:
This isn't user facing, but let's update this as well to modular pipeline and other micro-packages.